### ISSUE TITLE
[BUG] Posting draft returns error

### ISSUE THREAD

**Issue by Vyling** [MasterSteelblade/WaterfallBugsFeatures#58]
**Describe the bug**
I had a post in my drafts that I accidentally drafted yesterday. I tried to post it now, but instead I got the error message "There was an error trying to post. Please contact support."

**To Reproduce**
1. draft a reblog
2. Go to drafts
3. Try to publish 
4. Get error

**Expected behavior**
It should post the draft.

**Screenshots**
![image](https://user-images.githubusercontent.com/30179258/105876346-2eb75c80-5fff-11eb-94f0-0c2857fb68ef.png)
(in case there's any necessary info about date, user, type of post, etc.)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Vivaldi
 - Version latest

**Additional context**
This is the post I was trying to post: https://jellyenvy.waterfall.social/post/462771
On top of that, I drafted a text post just saying "test" before this, went into the edit state of it, pressed the back button, pressed edit on the draft that caused this bug and just pressed "post now". 
Maybe this helps if it's the workflow causing the problem.

**Comment by Vyling** [comment]
Could actually quickly reproduce it by just putting a reblog into drafts, edit that draft, and try to hit "post now".
I wanted to get this issue up before further testing in case it takes longer, but... well.

**Comment by Vyling** [comment]
It gets even weirder.
The post in question just posted to my blog without my intervention 
And it says that I reblogged myself.
I always only tried to reblog it by clicking on edit > post now 

As I originally tried to queue it.... maybe I actually did and it somehow still ended up in the drafts? Honestly I didn't check everything in my queue. But even then... why does it say that I reblogged myself? I'm so confused.
